backend/webdav/webdav.go

```diff
@@ -1350,9 +1413,105 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
+	hasher := md5.New()
+	_, err = hasher.Write([]byte(o.filePath()))
+	if err != nil {
+		return fmt.Errorf("chunked upload couldn't hash URL: %w", err)
+	}
+	uploadDir := "rclone-chunked-upload-" + hex.EncodeToString(hasher.Sum(nil))
+	fs.Debugf(src, "Starting multipart upload to temp dir %q", uploadDir)
+
+	opts := rest.Opts{
+		Method:     "MKCOL",
+		Path:       uploadDir + "/",
+		NoResponse: true,
+		RootURL:    o.fs.uploadURL,
+	}
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err := o.fs.srv.Call(ctx, &opts)
+		return o.fs.shouldRetry(ctx, resp, err)
+	})
+	if err != nil {
+		if apiErr, ok := err.(*api.Error); ok {
+			// Upload directory already exists, so ignore
+			if apiErr.StatusCode != http.StatusMethodNotAllowed || apiErr.Message != "The resource you tried to create already exists" {
+				return fmt.Errorf("making upload directory failed: %w", err)
+			}
+		}
+	}
+	defer atexit.OnError(&err, func() {
+		// Try to abort the upload, but ignore the error.
+		fs.Debugf(src, "Cancelling chunked upload")
+		_ = o.fs.Purge(ctx, uploadDir)
+	})()
+
+	var (
+		size         = src.Size()
+		uploadedSize = int64(0)
+		partObj      = &Object{
+			fs: o.fs,
+		}
+	)
+	for uploadedSize < size {
+		// Upload chunk
+		contentLength := int64(partObj.fs.opt.ChunkSize)
+		if size-uploadedSize < contentLength {
+			contentLength = size - uploadedSize
+		}
+		partObj.remote = fmt.Sprintf("%s/%015d-%015d", uploadDir, uploadedSize, uploadedSize+contentLength)
+		extraHeaders := map[string]string{}
+		// TODO: Fix broken retries. I suspect io.LimitReader() is the culprit here because it prevents it from being rewound.
+		// 2022-04-28 15:59:06 ERROR : stuff/video.avi: Failed to copy: uploading chunk failed: Put "https://censored.com/remote.php/dav/uploads/Admin/rclone-chunked-upload-censored/000006113198080-000006123683840": http2: Transport: cannot retry err [http2: Transport received Server's graceful shutdown GOAWAY] after Request.Body was written; define Request.GetBody to avoid this error
+
+		// TODO: support chunk upload retries.
+		// Problem: the mem usage can be too big if user configures it to a big value.
+		buf := make([]byte, partObj.fs.opt.ChunkSize)
+		in := readers.NewRepeatableLimitReaderBuffer(in0, buf, int64(partObj.fs.opt.ChunkSize))
+		//getBody := func() (io.ReadCloser, error) {
+		fs.Logf(nil, "body type: %T", in0)
+		// May need to do that in error callback: in.Seek(uploadedSize, io.SeekStart)
+		//return in, nil
+		//}
+
+		//body, _ := getBody()
+
+		err = partObj.updateSimple(ctx, in, nil, partObj.remote, contentLength, "application/x-www-form-urlencoded", extraHeaders, o.fs.uploadURL, options...)
+		if err != nil {
+			return fmt.Errorf("uploading chunk failed: %w", err)
+		}
+		uploadedSize += contentLength
+	}
+
+	// Finish
+	var resp *http.Response
+	opts = rest.Opts{
+		Method:     "MOVE",
+		Path:       o.fs.filePath(path.Join(uploadDir, ".file")),
```

Comment:
Had to change 
`Path: o.fs.filePath(path.Join(uploadDir, ".file"))`
to
`Path: path.Join(uploadDir, ".file")`
On my local branch for the final MOVE (assemble chunks) request to be successful. I wonder if this is because I have my webdav URL improperly formatted or something. I am assuming this line has worked for other people because it was in V1 of this MR as well.